lib/saluki-core/src/topology/interconnect/fixed_event_buffer.rs

```diff
@@ -0,0 +1,340 @@
+    where
+        F: Fn(&Event) -> bool,
+    {
+        let events = &mut self.data_mut().events;
+
+        let mut indices_to_remove = Vec::new();
+        let mut removed_events = VecDeque::new();
+
+        for (pos, event) in events.0.iter_mut().enumerate() {
+            if predicate(event) {
+                indices_to_remove.push(pos);
+            }
+        }
+
+        // Remove elements from back to front to avoid index shifting issues
+        for &pos in indices_to_remove.iter().rev() {
+            if pos < events.0.len() {
+                removed_events.push_back(events.0.swap_remove_back(pos).unwrap());
+            }
+        }
+
+        removed_events.into_iter()
+    }
+}
+
+impl fmt::Debug for FixedSizeEventBuffer {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("FixedSizeEventBuffer")
+            .field("event_len", &self.len())
+            .finish()
+    }
+}
+
+impl IntoIterator for FixedSizeEventBuffer {
+    type Item = Event;
+    type IntoIter = IntoIter;
+
+    fn into_iter(self) -> Self::IntoIter {
+        IntoIter { inner: self }
+    }
+}
+
+impl<'a> IntoIterator for &'a FixedSizeEventBuffer {
+    type Item = &'a Event;
+    type IntoIter = std::collections::vec_deque::Iter<'a, Event>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.data().events.0.iter()
+    }
+}
+
+impl<'a> IntoIterator for &'a mut FixedSizeEventBuffer {
+    type Item = &'a mut Event;
+    type IntoIter = std::collections::vec_deque::IterMut<'a, Event>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.data_mut().events.0.iter_mut()
+    }
+}
+
+pub struct IntoIter {
+    inner: FixedSizeEventBuffer,
+}
+
+impl Iterator for IntoIter {
+    type Item = Event;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.inner.data_mut().events.0.pop_front()
+    }
+}
+
+/*
+/// Helper trait for working with event buffer object po`ols.
+#[async_trait]
+trait FixedSizeEventBufferPoolExt {
+    /// Chunks a set of input events into fixed-size buffers, yielding them to a user-generated future.
+    ///
+    /// This method handles taking a set of input events and chunking them into fixed-sized event buffers, where each
+    /// event buffer is then passed to a user-generated future, created from the given closure. This makes it easy to
+    /// transform an unbounded set of input events into a set of fixed-size event buffers that can be forwarded to a
+    /// downstream component.
+    async fn chunked_send<I, F, Fut, E>(&self, events: I, send: F) -> Result<(), E>
+    where
+        Self: ObjectPool<Item = FixedSizeEventBuffer> + Sized,
+        I: IntoIterator<Item = Event> + Send,
+        I::IntoIter: Send,
+        F: Fn(FixedSizeEventBuffer) -> Fut + Send,
+        Fut: Future<Output = Result<(), E>> + Send,
+        E: Send;
+}
+
+#[async_trait]
+impl<T> FixedSizeEventBufferPoolExt for T
+where
+    T: ObjectPool<Item = FixedSizeEventBuffer>,
```

Comment:
Intentionally left?

Reply:
Ah, yeah. I had followed shortly after this comment with a commit to both remove the now-unused `EventBuffer`, and then also remove the commented-out code here. 😅